src/main/java/frc/robot/subsystems/PivotSparkMax.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.subsystems;
+
+import org.littletonrobotics.junction.Logger;
+
+import com.revrobotics.CANSparkMax;
+import com.revrobotics.AbsoluteEncoder;
+import com.revrobotics.SparkMaxPIDController;
+import com.revrobotics.CANSparkMax.ControlType;
+import com.revrobotics.CANSparkMaxLowLevel.MotorType;
+import com.revrobotics.SparkMaxAbsoluteEncoder.Type;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.Timer;
+import frc.robot.Constants.PivotConstants;
+
+public class PivotSparkMax implements PivotIO {
+  private final CANSparkMax leftMotor;
+  private final CANSparkMax rightMotor;
+  private final AbsoluteEncoder leftEncoder;
+  private final AbsoluteEncoder rightEncoder;
+  private final SparkMaxPIDController leftController;
+  private final SparkMaxPIDController rightController;
+  private final PivotIOInputsAutoLogged inputs = new PivotIOInputsAutoLogged();
+  /** Creates a new PivotReal. */
+  public PivotSparkMax(int leftMotorID, int rightMotorID) {
+    this.leftMotor = new CANSparkMax(leftMotorID, MotorType.kBrushless);
+    this.rightMotor = new CANSparkMax(rightMotorID, MotorType.kBrushless);
+    
+    this.leftEncoder = leftMotor.getAbsoluteEncoder(Type.kDutyCycle);
+    this.rightEncoder = rightMotor.getAbsoluteEncoder(Type.kDutyCycle);
+
+    this.leftController = leftMotor.getPIDController();
+    leftController.setP(PivotConstants.PID_P);
+    leftController.setI(PivotConstants.PID_I);
+    leftController.setD(PivotConstants.PID_D);
+    leftController.setFF(PivotConstants.PID_FF);
+    this.rightController = rightMotor.getPIDController();
+    rightController.setP(PivotConstants.PID_P);
+    rightController.setI(PivotConstants.PID_I);
+    rightController.setD(PivotConstants.PID_D);
+    rightController.setFF(PivotConstants.PID_FF);
+
+    Timer.delay(0.5);
+    leftMotor.burnFlash();
+    Timer.delay(0.5);
+    rightMotor.burnFlash();
+    Timer.delay(0.5);
+  }
+
+  /**
+   * Rotate both motors.
+   * @param speed The speed at which to rotate the motors.
+   */
+  public void rotatePivot(double speedLeft, double speedRight) {
+    leftController.setReference(speedLeft, ControlType.kDutyCycle);
+    rightController.setReference(speedRight, ControlType.kDutyCycle);
+  }
+
```

Comment:
You can add a feedPivotPID() method which is basically just to set the voltage of the motors based on calculated output from PID controller and feedforward. This is separate from the method to change the desired angle, but they will both be called in any pivot control method.